File: fortran/pr009.f90
! Problem 9
!
! 25 January 2002
!
! A Pythagorean triplet is a set of three natural numbers, a < b < c,
! for which,
!
! a^2 + b^2 = c^2
!
! For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
!
! There exists exactly one Pythagorean triplet for which a + b + c =
! 1000.
! Find the product abc.
! 31875000

! This particular problem is small enough that we can easily just do a brute
! force search.

subroutine pr009

  implicit none

  integer :: a, b, c

  do a = 1, 999
    do b = a, 1000 - a
      c = 1000 - a - b
      if (a*a + b*b == c*c) print *, (a*b*c)
    end do
  end do

end subroutine pr009

File: erlang/pr009.erl
% Problem 9
%
% 25 January 2002
%
%
% A Pythagorean triplet is a set of three natural numbers, a < b < c, for
% which,
%
% a^2 + b^2 = c^2
%
% For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
%
% There exists exactly one Pythagorean triplet for which a + b + c = 1000.
% Find the product abc.
%
% 31875000

-module(pr009).
-export([solve/0]).

solve() ->
	outer(1).

outer(1000) -> unknown;
outer(A) -> inner(A, A).

inner(A, 1000) -> outer(A + 1);
inner(A, B) ->
	C = 1000 - A - B,
	if
		C > B andalso A * A + B * B =:= C * C ->
			A * B * C;
		true ->
			inner(A, B+1)
	end.
